tempesta_fw/htype.h

```diff
@@ -0,0 +1,113 @@
+/**
+ *		Tempesta FW
+ *
+ * HTTP types: this file defines character types for HTTP like ctype.h
+ *
+ * Copyright (C) 2016 Tempesta Technologies, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License,
+ * or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59
+ * Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+ */
+#ifndef __HTYPE_H__
+#define __HTYPE_H__
+
+/**
+ * ASCII codes to accept HTTP token (RFC 7230 3.2.6).
+ */
+static const unsigned char token_a[] __attribute__((aligned(64))) = {
```

Comment:
Yes, the copies are desired. Since it's declared as `static` each translation unit will work with it's own copy and there won't be warnings. The data structure is just 4 cache lines, moreover only 2 of them are frequently used. Meantime, TFW_LC() is very performance critical call. When you make `static const` declaration compiler can do very good optimization work efficiently inlining constants to the code. If you define the constants table in other translation unit, then you lose the optimization opportunity.